src/modules/NewsList/index.tsx

```diff
@@ -224,29 +296,79 @@ export function NewsList(_props: NewsListProps): ReactElement {
                                     }
                                 }
                                 publishedDate
-                                source
                                 title
+                                ... on NewsContent {
+                                    newsSource {
+                                        id
+                                        name
+                                    }
+                                }
                             }
                         }
                     }
                 }
             }
         `,
+        mergeResults,
         requestPolicy: 'cache-and-network',
         variables: {
             filter: {
                 companyIds: state.company ? [state.company.id] : undefined,
                 contentTypes: [ContentType.News],
                 searchTerm: state.searchTerm,
             },
+            fromIndex: state.fromIndex,
+            size: listSize,
         },
     });
 
+    const onChangeSearch = useCallback<ChangeHandler<string>>(
+        (_event, change) => {
+            setState((s) => ({
+                ...s,
+                canRefetch: false,
+                fromIndex: 0,
+                lastRefetch: DateTime.now(),
+                searchTerm: change.value || '',
+            }));
+        },
+        [state]
+    );
+
+    const onLoadMore = useCallback<ChangeHandler<number>>(
```

Comment:
If this just passes the args through and doesn't use state, can you just pass the handler down directly?